src/types/game_time.rs

```diff
@@ -0,0 +1,384 @@
+        Self(red::GameTime { seconds })
+    }
+}
+
+impl From<GameTime> for u32 {
+    fn from(value: GameTime) -> Self {
+        value.0.seconds
+    }
+}
+
+impl IntoRepr for GameTime {
+    type Repr = u32;
+
+    fn into_repr(self) -> Self::Repr {
+        self.0.seconds
+    }
+}
+
+impl FromRepr for GameTime {
+    type Repr = u32;
+
+    fn from_repr(repr: Self::Repr) -> Self {
+        Self::from(repr)
+    }
+}
+
+#[cfg(feature = "time")]
+impl TryFrom<GameTime> for time::Time {
+    type Error = time::error::ComponentRange;
+
+    fn try_from(value: GameTime) -> Result<Self, Self::Error> {
+        Self::from_hms(
+            value.hour() as u8,
+            value.minute() as u8,
+            value.second() as u8,
+        )
+    }
+}
+
+#[cfg(feature = "time")]
+impl From<time::Time> for GameTime {
+    fn from(value: time::Time) -> Self {
+        Self::new(
+            0,
+            value.hour() as u32,
+            value.minute() as u32,
+            value.second() as u32,
+        )
+    }
+}
+
+#[cfg(feature = "time")]
+impl std::ops::Add<time::Time> for GameTime {
+    type Output = Self;
+
+    fn add(self, rhs: time::Time) -> Self::Output {
+        use std::ops::AddAssign;
+        let mut copy = self.clone();
+        copy.add_assign(rhs);
+        copy
+    }
+}
+
+#[cfg(feature = "time")]
+impl std::ops::AddAssign<time::Time> for GameTime {
+    fn add_assign(&mut self, rhs: time::Time) {
+        self.add_hours(rhs.hour() as u32);
+        self.add_minutes(rhs.minute() as u32);
+        self.add_seconds(rhs.second() as u32);
+    }
+}
+
+#[cfg(feature = "time")]
+impl std::ops::Sub<time::Time> for GameTime {
+    type Output = Self;
+
+    fn sub(self, rhs: time::Time) -> Self::Output {
+        use std::ops::SubAssign;
+        let mut copy = self.clone();
+        copy.sub_assign(rhs);
+        copy
+    }
+}
+
+#[cfg(feature = "time")]
+impl std::ops::SubAssign<time::Time> for GameTime {
+    fn sub_assign(&mut self, rhs: time::Time) {
+        self.sub_hours(rhs.hour() as u32);
+        self.sub_minutes(rhs.minute() as u32);
+        self.sub_seconds(rhs.second() as u32);
+    }
+}
+
+#[derive(Debug)]
+pub enum GameTimeError {
+    OutOfRange,
+}
+
+impl std::fmt::Display for GameTimeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Self::OutOfRange => "invalid GameTime: out-of-range",
+            }
+        )
+    }
+}
+
+impl std::error::Error for GameTimeError {}
+
+#[cfg(feature = "chrono")]
+pub fn cyberpunk_epoch() -> chrono::DateTime<chrono_tz::Tz> {
+    chrono_tz::US::Pacific
+        .with_ymd_and_hms(2077, 4, 16, 1, 24, 0)
+        .unwrap()
+}
+
+#[cfg(feature = "chrono")]
+impl From<GameTime> for chrono::DateTime<chrono::Utc> {
+    fn from(value: GameTime) -> Self {
+        // SAFETY: seconds being u32 it fits in i64, and nanos are zero
```

Comment:
```suggestion
        // seconds being u32 it fits in i64, and nanos are zero
```
safety comments are for code that can be unsafe in the Rust sense, e.g. invoke undefined behavior

Reply:
done in 5e832ec